ui/src/components/EdgePin.tsx

```diff
@@ -0,0 +1,47 @@
+import L, { LatLng, Point } from "leaflet";
+import * as React from "react";
+import { Marker } from "react-leaflet";
+import { EdgeType } from "./EdgePins";
+import { arrow } from "./Arrow";
+
+export interface EdgePinProps {
+  position: LatLng;
+  angle: number;
+  edgeType: EdgeType;
+}
+
+export function getAnchor(edgeType: EdgeType): Point {
+  switch (edgeType) {
+    case EdgeType.Top:
+      return new L.Point(0, -10);
+    case EdgeType.Bottom:
+      return new L.Point(0, 110);
+    case EdgeType.Left:
+      return new L.Point(-30, 0);
+    default:
+      return new L.Point(80, 0);
```

Comment:
Where are these numbers coming from? If they're just based on experiments, can we leave a comment explaining how we got there?